### PR TITLE
FFM-9085: Add a terraform resource to create a new target

### PR DESCRIPTION
**Pull request by ribeirophillipe**
## Describe your changes
* Implement CRUD actions for creating a new target in terraform
* Add all the interfaces and implementations
* 
## Comment Triggers

<details>
  <summary>PR Check triggers</summary>
  
- Build: `trigger build`
- Sub Category Field Check: `trigger subcategoryfieldcheck`

### CODE REVIEW

**Review by rohitk-h — COMMENTED** [review]
Please add unit tests for the resource. Ref https://harness.atlassian.net/wiki/spaces/PLATFORM/pages/21122646032/Terraform+Onboarding+and+Quickstart+for+Creating+Resource+in+Provider

**Comment by ribeirophillipe** [comment]
@rohitk-h unit tests were added!

**Review by rathodmeetsatish — CHANGES_REQUESTED** [review]
Remove makefile changes, tf_build.sh

**Comment by ribeirophillipe** [comment]
@rathodmeetsatish done

**Review by davejohnston — DISMISSED** [review]
This looks good. Ship it

File: internal/service/platform/user/resource_user.go [hunk 1]
```diff
@@ -104,7 +104,8 @@ func ResourceUser() *schema.Resource {
 							Type:        schema.TypeBool,
 							Optional:    true,
 						},
-					}},
+					},
```
Comment (rathodmeetsatish) [inline internal/service/platform/user/resource_user.go:107]: remove these changes

File: internal/service/cd/delegate/delegate_helper_test.go [hunk 2]
```diff
@@ -63,7 +64,7 @@ func deleteDelegate(t *testing.T, name string) {
 	cli, err := client.NewClientWithOpts(client.FromEnv)
 	require.NoError(t, err, "failed to create docker client: %s", err)
 
-	err = cli.ContainerStop(context.Background(), name, nil)
```
Comment (rathodmeetsatish) [inline internal/service/cd/delegate/delegate_helper_test.go:66]: why have we changed this revert if not needed

**Review by rathodmeetsatish — CHANGES_REQUESTED** [review]
generate docs and add changelog file

File: internal/service/platform/user/resource_user.go (hunk 1 above)
Comment (ribeirophillipe) [reply internal/service/platform/user/resource_user.go:107]: done

File: internal/service/cd/delegate/delegate_helper_test.go (hunk 2 above)
Comment (ribeirophillipe) [reply internal/service/cd/delegate/delegate_helper_test.go:66]: it gives a compilation error ...

File: .changelog/670.txt [hunk 3]
```diff
@@ -1,3 +1,7 @@
+```release-note:new-resource
```
Comment (rathodmeetsatish) [inline .changelog/670.txt:1]: add a file corresponding to your pr number which is 660

File: examples/data-sources/harness_platform_monitored_service/data-source.tf [hunk 4]
```diff
@@ -1,5 +1,5 @@
 data "harness_platform_monitored_service" "example" {
   identifier = "identifier"
-  org_id = "org_id"
```
Comment (rathodmeetsatish) [inline examples/data-sources/harness_platform_monitored_service/data-source.tf:3]: remove changes which dont belong to ff

File: examples/data-sources/harness_platform_slo/data-source.tf [hunk 5]
```diff
@@ -1,5 +1,5 @@
 data "harness_platform_slo" "example" {
   identifier = "identifier"
-  org_id = "org_id"
```
Comment (rathodmeetsatish) [inline examples/data-sources/harness_platform_slo/data-source.tf:3]: remove changes which dont belong to ff

File: examples/resources/harness_platform_policyset/resource.tf [hunk 6]
```diff
@@ -17,8 +17,8 @@ resource "harness_platform_policyset" "test" {
   action     = "onrun"
   type       = "pipeline"
   enabled    = true
-  org_id      = "terraform_example_org"
-  project_id  = "terraform_test_project"
+  org_id     = "terraform_example_org"
```
Comment (rathodmeetsatish) [inline examples/resources/harness_platform_policyset/resource.tf:20]: remove changes which dont belong to ff

File: examples/resources/harness_platform_slo/resource.tf [hunk 7]
```diff
@@ -1,14 +1,14 @@
 resource "harness_platform_slo" "example" {
-  org_id = "org_id"
+  org_id     = "org_id"
   project_id = "project_id"
   identifier = "identifier"
```
Comment (rathodmeetsatish) [inline examples/resources/harness_platform_slo/resource.tf:4]: remove changes which dont belong to ff

File: internal/service/cd/delegate/delegate_helper_test.go (hunk 2 above)
Comment (rathodmeetsatish) [inline internal/service/cd/delegate/delegate_helper_test.go:66]: remove changes which dont belong to ff

**Review by rathodmeetsatish — CHANGES_REQUESTED** [review]

File: docs/resources/platform_feature_flag_target.md [hunk 8]
```diff
@@ -0,0 +1,46 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "harness_platform_feature_flag_target Resource - terraform-provider-harness"
+subcategory: ""
```
Comment (rathodmeetsatish) [inline docs/resources/platform_feature_flag_target.md:4]: subcategory: "NextGen"

**Review by rathodmeetsatish — CHANGES_REQUESTED** [review]

**Review by rathodmeetsatish — APPROVED** [review]

**Pull request merged**